java/ql/test/library-tests/dataflow/range-analysis-inline/range.ql

```diff
@@ -0,0 +1,37 @@
+/**
+ * Inline range analysis tests for Java.
+ * See `shared/util/codeql/dataflow/test/InlineFlowTest.qll`
+ */
+
+import java
+import semmle.code.java.dataflow.RangeAnalysis
+private import codeql.util.test.InlineExpectationsTest
+private import TestUtilities.internal.InlineExpectationsTestImpl
+private import Make<Impl> as IET
+
+module RangeTest implements IET::TestSig {
+  string getARelevantTag() { result = "bound" }
+
+  predicate hasActualResult(Location location, string element, string tag, string value) {
+    tag = "bound" and
+    exists(Expr e, int lower, int upper |
+      constrained(e, lower, upper) and
+      e instanceof VarRead and
+      e.getCompilationUnit().fromSource()
+    |
+      location = e.getLocation() and
+      element = e.toString() and
+      if lower = upper
+      then value = "\"" + e.toString() + " = " + lower.toString() + "\""
+      else
+        value = "\"" + e.toString() + " in [" + lower.toString() + ".." + upper.toString() + "]\""
+    )
+  }
+
+  private predicate constrained(Expr e, int lower, int upper) {
+    lower = min(int delta | bounded(e, any(ZeroBound z), delta, false, _)) and
+    upper = min(int delta | bounded(e, any(ZeroBound z), delta, true, _))
```

Comment:
The library is expected to only report the best bounds, so the `min` is unnecessary.
```suggestion
    bounded(e, any(ZeroBound z), lower, false, _) and
    bounded(e, any(ZeroBound z), upper, true, _)
```

Reply:
Oh, it will already choose the tightest one? Ah, yes, it says _and_  `bestBound(..)`.